das/state.go

```diff
@@ -0,0 +1,114 @@
+package das
+
+import (
+	"sync"
+	"sync/atomic"
+
+	"github.com/celestiaorg/celestia-node/header"
+)
+
+// state collects information about the DASer process. Currently, there are
+// only two sampling routines: the main sampling routine which performs sampling
+// over current network headers, and the `catchUp` routine which performs sampling
+// over past headers from the last sampled checkpoint.
+type state struct {
+	sampleStateLk sync.RWMutex
+	sampleState   RoutineState // tracks information related to the main sampling routine
+
+	catchUpStateLk sync.RWMutex
+	catchUpState   RoutineState // tracks information related to the `catchUp` routine
+}
+
+// RoutineState contains important information about the state of a
+// current sampling routine.
+type RoutineState struct {
+	// tracks whether routine is running
+	IsRunning uint64 `json:"is_running"`
```

Comment:
You can use https://github.com/uber-go/atomic/blob/master/bool.go instead of uint64